hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/Invocation.java

```diff
@@ -580,7 +578,7 @@ private void handleInvocationException(Exception e) {
 
     private void notifyWithExceptionWhenTargetIsNull() {
         ClusterState clusterState = context.clusterService.getClusterState();
-        if (clusterState == FROZEN || clusterState == PASSIVE) {
+        if (!clusterState.isMigrationAllowed()) {
             notifyError(new IllegalStateException("Partitions can't be assigned since cluster-state: " + clusterState));
```

Comment:
When migration is not allowed, even in `IN_TRANSITION` state, target may be null. Because partitions won't be assigned. 

Exception message is a little misleading. I'll change that to 

```java
new IllegalStateException("Target of invocation cannot be found! Partition owner is null "
                    + "but partitions can't be assigned in cluster-state: " + clusterState);
```